test/net-api/test-connection.js

```diff
@@ -0,0 +1,114 @@
+var tape = require('tape')
+var utp = require('../../index.js')
+
+tape('net-api connection', function (t) {
+  t.test('connect', function (t) {
+    var connected = false
+
+    var server = utp.createServer(function () {
+      connected = true
+    })
+
+    server.listen(0, function () {
+      var socket = utp.connect(server.address().port)
+
+      socket.on('connect', function () {
```

Comment:
Now, by only calling connect() when the server is already listening, I get a UTP_ETIMEOUT, still investigating why, @mafintosh ideas?